### ISSUE TITLE
I do not understand where the lags come from

### ISSUE THREAD

**Issue by Choumpsi** [mariotaku/moonlight-tv#137]
## Basic information

 - Device model: LG OLED65C1
 - Firmware version: 03.21.05
 - webOS version: 6.2.0
 - Moonlight version: 1.4.0
 - Is audio working: Lagging with video
 - Is video working: Lagging in all resolutions
 - Is input working: Yes

## Description
Hello, I hope I'm posting at the right place !
I have Moonlight installed on my LG OLED65C1 TV, which is connected to the router via Ethernet, same with my computer.
Despite the computer running the games smoothly no matter the resolution (same in 720p, 1080p or 1440p), the stream on the TV is constantly lagging like 2 times every second. My internet connection is about 400Mbps, so I don't really understand where they could come from.

I've tried different video encoders but it doesn't change anything.

Thank you in advance for any help, do not hesitate to tell me how I can provide any logs if necessary.

**Comment by Choumpsi** [comment]
The TV is connected to the router via Ethernet, and the computer is connected via Ethernet as well.

**Comment by mariotaku** [comment]
It sounds strange, but did you try to connect your PC to router via wi-fi instead?

**Comment by mariotaku** [comment]
Some reports severe packet drop while on Ethernet. You can confirm by open performance stats with long pressing back button.

**Comment by Choumpsi** [comment]
Yes, I tried both combinations.

For this example I have Moonlight settings on  : 
-  1080p - 60FPS - 20000kbps bitrate
-  No optimization for streaming
-  Video decoder running on NDL
-  Preference for the H265 codec

In case it may help, my computer is running the game on 1080p with : 
-  16GB of RAM
-  GeForce GTX 1080Ti
-  I7-8700K

Having the PC connected via wi-fi I have : 
-   Network RTT of approx. 34ms
-   Network frame drop of approx. 15%

Connecting via Ethernet, only RTT drops to 2-3ms. The rest stays the same.
In both cases, the decode time is about 0.3ms.
The app constantly says "Unstable connection", maybe it is a TV-side problem I don't know.

Thanks for your help.

**Comment by Bruffandino** [comment]
Try forcing the host's ethernet adapter to 100Mbps. It's annoying, but it fixed the problem for me.

**Comment by Papabeer323** [comment]
I had the same issue. Doing as Bruffandino said fixed it for me as well.

> Try forcing the host's ethernet adapter to 100Mbps. It's annoying, but it fixed the problem for me.

**Comment by joaocaparroz** [comment]
Same problem here, but my TV is a LG OLED55C1. Tested different resolutions, codecs, ethernet adapter speeds. Nothing fixed it.

**Comment by GaspardMonier** [comment]
Same poblem, I also have an OLED55C1 connected to wifi 6
My PC is connected via ethernet

rtx 2070 super
ryzen 3700x
32gb ram

**Comment by germsb** [comment]
Same problem for me.
OLED48C1 wifi ac --> router <-- PC rj45 GbE   (250 mb/s)

rtx 3070 ti
ryzen 5800x
32gb ram

Works well over wifi.

**Comment by bluefantail** [comment]
Any update on this issue?

I hadn't used moonlight in a while but fired up the WebOS TV app tonight to have a go and ran into what I think is this problem, really laggy video that completely freezes up and displays 'Unstable video' in the top right corner. Updated to the latest moonlight version, PC video card drivers etc, and messed about with changing client decoder types and quality settings - but couldn't get anything to work until I found the note here about Host link speed.

Switched Speed & Duplex in my ethernet adapter settings from 'Auto Negotiation' (which always picks 1gbps) to '100 Mbps Full Duplex' - all of a sudden streams flawlessly like it used to.

Setup is Windows PC -> WebOS client on an LG C1, both wired to MikroTik hAP ac3 router via ethernet.

Obviously this workaround works for some folk, but seems real strange and unideal switching adapter speeds every time :)

**Comment by mariotaku** [comment]
What about a USB NIC? Not so ideal either but that's an issue that can't be solved on the app side...

**Comment by bluefantail** [comment]
> What about a USB NIC? Not so ideal either but that's an issue that can't be solved on the app side...

Ah so are you suggesting this like a know limitation? Something changed in the way NVIDIA gamestream is working on the host or something? Just confused because I've used for hours on the same hardware and now it's gone weird. I have noticed the C1 had a software update recently though as well

**Comment by mariotaku** [comment]
> > What about a USB NIC? Not so ideal either but that's an issue that can't be solved on the app side...
> 
> Ah so are you suggesting this like a known limitation? Something changed in the way NVIDIA gamestream is working on the host or something? Just confused because I've used for hours on the same hardware and now it's suddenly gone weird. I have noticed the C1 had a software update recently though as well

According to some reports, LG TV have 100mb NIC and when you connect it to 1gbps NIC it will have bad performance. I'm not quite sure because I'm connecting with a 1gbps switch right now...

**Comment by bluefantail** [comment]
> > > What about a USB NIC? Not so ideal either but that's an issue that can't be solved on the app side...
> > 
> > 
> > Ah so are you suggesting this like a known limitation? Something changed in the way NVIDIA gamestream is working on the host or something? Just confused because I've used for hours on the same hardware and now it's suddenly gone weird. I have noticed the C1 had a software update recently though as well
> 
> According to some reports, LG TV have 100mb NIC and when you connect it to 1gbps NIC it will have bad performance. I'm not quite sure because I'm connecting with a 1gbps switch right now...

Yeah the LG C1 only has 100 Mpbs (I think pretty standard for most or all LG TV's). I don't know enough about networking to understand why the host having a different rate would matter though

**Comment by mwLabs-eu** [comment]
Hi, it seems, this is still a thing. Any chance, this will get fixed? Quite annoying to change the adapters speed every time when playing.
